url85940d3d8e_ccp999_Expr4j_tgz-pJ8-org_boris_expr_util_ExprEvaluatorJ8/src/org/boris/expr/util/Condition.java

```diff
@@ -0,0 +1,110 @@
+/*******************************************************************************
+ * This program and the accompanying materials
+ * are made available under the terms of the Common Public License v1.0
+ * which accompanies this distribution, and is available at 
+ * http://www.eclipse.org/legal/cpl-v10.html
+ * 
+ * Contributors:
+ *     Peter Smith
+ *******************************************************************************/
+package org.boris.expr.util;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.boris.expr.AbstractBinaryOperator;
+import org.boris.expr.Expr;
+import org.boris.expr.ExprBoolean;
+import org.boris.expr.ExprDouble;
+import org.boris.expr.ExprEqual;
+import org.boris.expr.ExprException;
+import org.boris.expr.ExprGreaterThan;
+import org.boris.expr.ExprGreaterThanOrEqualTo;
+import org.boris.expr.ExprInteger;
+import org.boris.expr.ExprLessThan;
+import org.boris.expr.ExprLessThanOrEqualTo;
+import org.boris.expr.ExprMissing;
+import org.boris.expr.ExprNotEqual;
+import org.boris.expr.ExprString;
+import org.boris.expr.function.excel.AND;
+
+public class Condition
+{
+    private List<AbstractBinaryOperator> operators = new ArrayList();
+    private AND func = new AND();
+
+    public static Condition valueOf(Expr arg) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 11. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.